tools/SourceKit/tools/sourcekitd/lib/Service/Requests.cpp

```diff
@@ -450,14 +450,35 @@ getInputBufForRequestOrEmitError(const RequestDict &Req,
   return buf;
 }
 
-/// Get 'key.sourcefile' value as a string. If it's missing, reply an error to
-/// \p Rec and return None.
+/// Get 'key.primary_file' value as a string. If it's missing, reply with an
+/// error and return \c None.
+///
+/// Fallsback to 'key.sourcefile' for compatibility.
 static Optional<StringRef>
-getSourceFileNameForRequestOrEmitError(const RequestDict &Req,
-                                       ResponseReceiver Rec) {
+getPrimaryFileForRequestOrEmitError(const RequestDict &Req,
+                                    ResponseReceiver Rec) {
+  Optional<StringRef> PrimaryFile = Req.getString(KeyPrimaryFile);
+  if (!PrimaryFile) {
+    // Fallback to the old key.sourcefile
+    PrimaryFile = Req.getString(KeySourceFile);
+    if (!PrimaryFile) {
+      Rec(createErrorRequestInvalid("missing 'key.primary_file'"));
+    }
+  }
+  return PrimaryFile;
+}
+
+/// Get 'key.primary_file' value as a string. If it's missing, reply with an
+/// error and return \c None.
+///
+/// Fallsback to 'key.sourcefile' for compatibility.
+static Optional<StringRef>
+getSourceFileForRequestOrEmitError(const RequestDict &Req,
```

Comment:
I don’t think the implementation matches the doc comment.

Reply:
I may or may not have copy pasted it :P